### PR TITLE
Add support for estimating render time in appleseed.cli

### PR DESCRIPTION
**Pull request by achalpandeyy**
As the first step towards issue #2438.

I initially submitted these changes in #2453  but realized that the indentation was off so I tried deleting the PR a bunch of times which made a big mess.
Also, the travis-ci build failed most probably due to some network issues which only added to the mess.
So I decided to close that PR and open a new one, hope you don't mind!

### CODE REVIEW

File: src/appleseed.cli/progresstilecallback.cpp [hunk 1]
```diff
@@ -90,12 +102,28 @@ namespace
...
+            // Keep track of the total number of rendered tiles.
+            m_rendered_tiles++;
+
+            // Retrieve the total number of tiles in the frame.
+            const size_t total_tiles = frame->image().properties().m_tile_count;
+
+            // Estimate remaining render time.
+            m_stopwatch.measure();
+            const double elapsed_time = m_stopwatch.get_seconds();
+            const double estimated_time = (elapsed_time / m_rendered_tiles) * (total_tiles - m_rendered_tiles);
```
Comment (dictoon) [inline src/appleseed.cli/progresstilecallback.cpp]: Maybe `remaining_time` is clearer?

File: src/appleseed.cli/progresstilecallback.cpp [hunk 2]
```diff
@@ -90,12 +102,28 @@ namespace
...
+
+            // Print estimated time.
+            LOG_INFO(m_logger, "remaining time %.3lf seconds", estimated_time);
```
Comment (dictoon) [inline src/appleseed.cli/progresstilecallback.cpp]: `%.3f` would have been enough here. That said, you should use `pretty_time()` which will print time in a more friendly manner than just seconds. Finally, I don't think we need subsecond precision for time estimations (they are very approximate anyway).

I would also prefer that both the progress message and the remaining time are printed on the same line, something like

    rendering, 53% done; about 5 hours, 30 minutes and 21 seconds remaining...

`pretty_time()` will take care of the latter part.

Btw, make sure to call `c_str()` on the string returned by `pretty_time()`.

File: src/appleseed.cli/progresstilecallback.cpp [hunk 3]
```diff
@@ -90,12 +102,28 @@ namespace
...
         }
 
       private:
         Logger&             m_logger;
         boost::mutex        m_mutex;
         size_t              m_rendered_pixels;
+        size_t              m_rendered_tiles;
+        Stopwatch<DefaultWallclockTimer>    m_stopwatch;
```
Comment (dictoon) [inline src/appleseed.cli/progresstilecallback.cpp]: Align members vertically.

You need to include `foundation/platform/defaulttimers.h` for `DefaultWallclockTimer`.

File: src/appleseed.cli/progresstilecallback.cpp (hunk 2 above)
Comment (achalpandeyy) [reply src/appleseed.cli/progresstilecallback.cpp]: done :)

File: src/appleseed.cli/progresstilecallback.cpp (hunk 3 above)
Comment (achalpandeyy) [reply src/appleseed.cli/progresstilecallback.cpp]: sorry didn't see it! gonna include it in next commit

Comment (achalpandeyy) [reply src/appleseed.cli/progresstilecallback.cpp]: done

File: src/appleseed.cli/progresstilecallback.cpp [hunk 4]
```diff
@@ -89,13 +102,33 @@ namespace
             const size_t total_pixels = frame->image().properties().m_pixel_count;
 
             // Print a progress message.
-            LOG_INFO(m_logger, "rendering, %s done", pretty_percent(m_rendered_pixels, total_pixels).c_str());
+            //LOG_INFO(m_logger, "rendering, %s done", pretty_percent(m_rendered_pixels, total_pixels).c_str());
```
Comment (dictoon) [inline src/appleseed.cli/progresstilecallback.cpp]: Don't forget to remove dead code.

Comment (achalpandeyy) [reply src/appleseed.cli/progresstilecallback.cpp]: I keep missing things for some reason! This time it is perfect.

**Comment by achalpandeyy** [comment]
Thanks for the merge! :)

**Pull request merged**